### PR TITLE
Set pdnsutil default editor to vi if editor is not found in path and $EDITOR is not set. 

### PR DESCRIPTION
**Pull request by hnsk**

### CODE REVIEW

**Comment by zeha** [comment]
This breaks setups that have `emacsclient -c` in EDITOR.

**Comment by cmouse** [comment]
After review this should be one commit.

File: pdns/pdnsutil.cc [hunk 1]
```diff
@@ -1001,15 +1001,19 @@ int editZone(DNSSECKeeper& dk, const DNSName &zone) {
...
-  cmdline=editor+" ";
+  cmdline=" ";
   if(gotoline > 0)
     cmdline+="+"+std::to_string(gotoline)+" ";
   cmdline += tmpnam;
-  int err=system(cmdline.c_str());
+  int err=system((editor + cmdline).c_str());
   if(err) {
-    unixDie("Editing file with: '"+cmdline+"', perhaps set EDITOR variable");
+    err=system(("vi" + cmdline).c_str());
```
Comment (cmouse) [inline pdns/pdnsutil.cc]: make "vi" a const class member or define

**Comment by Habbie** [comment]
@cmouse this makes me realise our CONTRIBUTING.md might need a note on squashing

File: pdns/pdnsutil.cc [hunk 2]
```diff
@@ -1001,15 +1001,19 @@ int editZone(DNSSECKeeper& dk, const DNSName &zone) {
...
+  }
+  if(err) {
+    throw runtime_error("Failed to edit with '"+editor+"' or 'vi': please set the EDITOR variable");
   }
   cmdline.clear();
+  editor.clear();
```
Comment (cmouse) [inline pdns/pdnsutil.cc]: pointless?

File: pdns/pdnsutil.cc (hunk 1 above)
Comment (zeha) [inline pdns/pdnsutil.cc]: why?

Comment (Habbie) [inline pdns/pdnsutil.cc]: Silently falling back to vi when the editor exits with a failure reason does not make sense.

**Comment by Habbie** [comment]
If you resolve the conflicts, we promise we will test this.

**Comment by hnsk** [comment]
I have no idea what I did over a year ago and can't be bothered figuring it out.

**Comment by slowaways** [comment]
Im sorry, but this PR was added in which version?

**Comment by zeha** [comment]
No version, this didn't get merged.

**Comment by slowaways** [comment]
I have the error:

```
sh: editor: command not found
Error: Editing file with: 'editor / tmp / pdnsutil-5VVukw', perhaps set EDITOR variable: Resource temporarily unavailable
```

And I can't solve it!

**Pull request closed without merging**